source/common/stats/fake_symbol_table_impl.h

```diff
@@ -75,22 +110,38 @@ class FakeSymbolTableImpl : public SymbolTable {
   void debugPrint() const override {}
 #endif
 
+  StoragePtr copyToBytes(absl::string_view name) override {
+    auto bytes = std::make_unique<Storage>(name.size() + StatNameSizeEncodingBytes);
+    uint8_t* buffer = saveLengthToBytesReturningNext(name.size(), bytes.get());
+    memcpy(buffer, name.data(), name.size());
+    return bytes;
+  }
+
+  void callWithStringView(StatName stat_name,
+                          const std::function<void(absl::string_view)>& fn) const override {
+    fn(toStringView(stat_name));
+  }
+
 private:
-  SymbolEncoding encodeHelper(absl::string_view name) const {
-    SymbolEncoding encoding;
-    encoding.addStringForFakeSymbolTable(name);
-    return encoding;
+  // Saves the specified length into the byte array, returning the next byte.
+  // There is no guarantee that bytes will be aligned, so we can't cast to a
+  // uint16_t* and assign, but must individually copy the bytes.
+  static uint8_t* saveLengthToBytesReturningNext(uint64_t length, uint8_t* bytes) {
```

Comment:
What do you think of the signature `writeLength(uint64_t length, uint8_t* dest)`? I was confused at first because sometimes we pass `bytes` as something being written, not a space in which writing occurs. The returned ptr could be explicit in the comment, but it's familiar enough from operations in <unordered_map> that it doesn't need to be explicit in the name.

Reply:
writeLengthReturningNext works for me.